src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,202 @@
+        Car car = createCarWithDefaultConstructor();
+
+        // when
+        Method[] declaredMethods = Car.class.getDeclaredMethods();
+        List<Method> methods = Arrays.stream(declaredMethods)
+                                     .filter(method -> method.getName().startsWith(PREFIX)).toList();
+        methods.forEach(method -> invokeMethod(method, car));
+
+        // then
+        assertAll(
+            () -> assertThat(methods).hasSize(2),
+            () -> assertThat(methods.stream().map(Method::getName).sorted()).containsExactly(
+                "testGetName", "testGetPrice"
+            )
+        );
+    }
+
+    @Test
+    @DisplayName("PrintView Annotation method 실행")
+    void invokePrintViewAnnotationMethod() {
+        // given
+        Car car = createCarWithDefaultConstructor();
+
+        Method[] declaredMethods = Car.class.getDeclaredMethods();
+        List<Method> methods = Arrays.stream(declaredMethods)
+                                     .filter(method -> method.isAnnotationPresent(PrintView.class)).toList();
+        methods.forEach(method -> invokeMethod(method, car));
+
+        // then
+        assertAll(
+            () -> assertThat(methods).hasSize(1),
+            () -> assertThat(methods.stream().map(Method::getName).toList()).containsExactly(
+                "printView"
+            )
+        );
+    }
+
+
+    @Test
+    @DisplayName("private field에 값 할당")
+    void privateField() {
+        // given
+        Car car = createCarWithDefaultConstructor();
+        final String NAME_TEST_VALUE = "test";
+        final int PRICE_TEST_VALUE = 111;
+
+        Field[] declaredFields = Car.class.getDeclaredFields();
+        List<Field> privateFields = Arrays.stream(declaredFields)
+                                          .filter(it -> it.getModifiers() == Modifier.PRIVATE)
+                                          .toList();
+        privateFields.forEach(
+            field -> {
+                field.setAccessible(true);
+                if (field.getType() == String.class) {
+                    fillStringArgument(car, field, NAME_TEST_VALUE);
+
+                } else if (field.getType() == int.class) {
+                    fillIntArgument(car, field, PRICE_TEST_VALUE);
+                }
+            }
+        );
+
+        assertAll(
+            () -> assertThat(getNameField(car)).isEqualTo(NAME_TEST_VALUE),
+            () -> assertThat(getPriceField(car)).isEqualTo(PRICE_TEST_VALUE)
+        );
+    }
+
+    @Test
+    void constructorWithArgs() {
+        Class<Car> carClass = Car.class;
+        final String NAME_TEST_VALUE = "test";
+        final int PRICE_TEST_VALUE = 111;
+        Constructor<?> constructorWithTwoParameter = Arrays.stream(carClass.getDeclaredConstructors())
+                                                           .filter(it -> it.getParameterCount() == 2)
+                                                           .findFirst().get();
```

Comment:
![스크린샷 2023-10-17 오전 10 00 45](https://github.com/next-step/jpa-query-builder/assets/12059592/2b3016e9-3b96-418e-a746-ccb82fe67812)

`findFirst`은 `Optional` 을 반환합니다. 이에 대한 처리를 해주시면 더 좋을것 같아요!
```suggestion
      .findFirst()
      .orElseThrow(() -> ...);
```